app/src/main/kotlin/com/google/samples/apps/nowinandroid/ui/interests2pane/InterestsListDetailScreen.kt

```diff
@@ -0,0 +1,140 @@
+import androidx.compose.material3.adaptive.ListDetailPaneScaffoldRole
+import androidx.compose.material3.adaptive.PaneAdaptedValue
+import androidx.compose.material3.adaptive.ThreePaneScaffoldNavigator
+import androidx.compose.material3.adaptive.rememberListDetailPaneScaffoldNavigator
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.navigation.NavGraphBuilder
+import androidx.navigation.NavType
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import androidx.navigation.navArgument
+import com.google.samples.apps.nowinandroid.feature.interests.InterestsRoute
+import com.google.samples.apps.nowinandroid.feature.interests.navigation.INTERESTS_ROUTE
+import com.google.samples.apps.nowinandroid.feature.interests.navigation.TOPIC_ID_ARG
+import com.google.samples.apps.nowinandroid.feature.topic.navigation.TOPIC_ROUTE
+import com.google.samples.apps.nowinandroid.feature.topic.navigation.navigateToTopic
+import com.google.samples.apps.nowinandroid.feature.topic.navigation.topicScreen
+
+private const val DETAIL_PANE_NAVHOST_ROUTE = "detail_pane_route"
+
+fun NavGraphBuilder.interestsListDetailScreen() {
+    composable(
+        route = INTERESTS_ROUTE,
+        arguments = listOf(
+            navArgument(TOPIC_ID_ARG) {
+                type = NavType.StringType
+                defaultValue = null
+                nullable = true
+            },
+        ),
+    ) {
+        InterestsListDetailScreen()
+    }
+}
+
+@Composable
+internal fun InterestsListDetailScreen(
+    viewModel: Interests2PaneViewModel = hiltViewModel(),
+) {
+    val selectedTopicId by viewModel.selectedTopicId.collectAsStateWithLifecycle()
+    InterestsListDetailScreen(
+        selectedTopicId = selectedTopicId,
+        onTopicClick = viewModel::onTopicClick,
+    )
+}
+
+@OptIn(ExperimentalMaterial3AdaptiveApi::class)
+@Composable
+internal fun InterestsListDetailScreen(
+    selectedTopicId: String?,
+    onTopicClick: (String) -> Unit,
+) {
+    val listDetailNavigator = rememberListDetailPaneScaffoldNavigator<Nothing>()
+    BackHandler(listDetailNavigator.canNavigateBack()) {
+        listDetailNavigator.navigateBack()
+    }
+
+    val nestedNavController = rememberNavController()
+
+    fun onTopicClickShowDetailPane(topicId: String) {
+        onTopicClick(topicId)
+        nestedNavController.navigateToTopic(topicId) {
+            popUpTo(DETAIL_PANE_NAVHOST_ROUTE)
+        }
+        listDetailNavigator.navigateTo(ListDetailPaneScaffoldRole.Detail)
+    }
+
+    ListDetailPaneScaffold(
+        scaffoldState = listDetailNavigator.scaffoldState,
+        listPane = {
+            InterestsRoute(
```

Comment:
Can we wrap both the `listPane` content and the `detailPane` content in `AnimatedPane(Modifier)`?

That should give us animations for swapping between list and detail.

Reply:
We cannot. Or rather, we can only use it for the list pane, which doesn't improve things much.

The problem is that `AnimatedPane` skips its content when the pane is hidden, so if you have a device which shows only one pane, the call to `navigate` on the nested `NavController` happens before the `NavHost` has been composed, causing a crash: [animated_pane_crash.txt](https://github.com/android/nowinandroid/files/14450438/animated_pane_crash.txt)

Reply:
Tracing this issue internally - I vote we merge this now to iterate upon, and update to new versions of the library.